frontend/app/routes/$lang+/_protected+/alerts+/subscribe+/confirm.tsx

```diff
@@ -44,17 +47,45 @@ export async function loader({ context: { session }, params, request }: LoaderFu
   const meta = { title: t('gcweb:meta.title.template', { title: t('alerts:confirm.page-title') }) };
 
   instrumentationService.countHttpStatus('alerts.confirm', 302);
-  return json({ csrfToken, meta }); //TODO get the language and email address entered by the user when they entered their information on the index route...
+
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const alertSubscription = await getSubscriptionService().getSubscription(userInfoToken.sin ?? '');
+  session.set('alertSubscription', alertSubscription);
+
+  return json({ csrfToken, meta, alertSubscription, userInfoToken }); //TODO get the language and email address entered by the user when they entered their information on the index route...
 }
 
 export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
   const formData = await request.formData();
+  //const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  //const alertSubscription = session.get('alertSubscription');
   const action = formData.get('action');
+  const instrumentationService = getInstrumentationService();
+  //const t = await getFixedT(request, handle.i18nNamespaces);
+  const formDataSchema = z.object({
+    confirmationCode: z.string().trim().max(100).optional(),
+  });
+  const data = {
+    confirmationCode: formData.get('confirmationCode') ? String(formData.get('confirmationCode')) : undefined,
+  };
+
+  const parsedDataResult = formDataSchema.safeParse(data);
+  if (!parsedDataResult.success) {
+    instrumentationService.countHttpStatus('alerts.confirm', 400);
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof formDataSchema>>,
+    });
+  }
+
   if (action === ConfirmSubscriptionCode.NewCode) {
     //TODO implement the code to request a new code and link that new code to the clients profile
   }
   if (action === ConfirmSubscriptionCode.Submit) {
     //TODO Validate the entered code and complete the user's registration to the alert me service if the code is correct
+    //COMMENTING THOSE LINES OF CODE TO PREVENT TEAM CITY FROM FAILING...
+    //const response = await getSubscriptionService().validateConfirmationCode(alertSubscription?.email ?? '', parsedDataResult.data.confirmationCode ?? '', userInfoToken.sin ?? '');
+    //const responseBody = await response.text();
```

Comment:
I feel we should be calling the `validateConfirmationCode(..)` function here. Any reason why the TeamCity build fails when these lines are added?

Reply:
If i remove the comment, npm run lint will fail and this will make the teamcity fail im afraid
![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/78967586/07197f75-18a8-4fef-bd87-b426a9d59504)

Reply:
service call has been kept, the variables declaration has been removed